bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableTableAdminClientWrapper.java

```diff
@@ -306,4 +317,109 @@ public Void apply(Empty input) {
   public ApiFuture<Operation> createTableFromSnapshotAsync(CreateTableFromSnapshotRequest request) {
     return ApiFutureUtil.adapt(delegate.createTableFromSnapshotAsync(request));
   }
+
+  @Override
+  public ApiFuture<Backup> createBackupAsync(CreateBackupRequest request) {
+    return ApiFutureUtil.transformAndAdapt(
+        delegate.createBackupAsync(
+            request.toProto(instanceName.getProjectId(), instanceName.getInstanceId())),
+        new Function<Operation, Backup>() {
+          @Override
+          public Backup apply(Operation operation) {
+            try {
+              return Backup.fromProto(
+                  operation.getResponse().unpack(com.google.bigtable.admin.v2.Backup.class));
+            } catch (InvalidProtocolBufferException e) {
+              return Backup.fromProto(com.google.bigtable.admin.v2.Backup.getDefaultInstance());
```

Comment:
Here & restoreTableAsync:

This looks a bit scary. Can you add comments when this would happen?

Reply:
ping